match_tpl.h

```diff
@@ -89,50 +117,80 @@ int32_t LONGEST_MATCH(deflate_state *const s, Pos cur_match) {
          * is limited to the lookahead, so the output of deflate is not
          * affected by the uninitialized values.
          */
-        cont = 1;
-        do {
-            match = window + cur_match;
-            if (LIKELY(*(bestcmp_t *)(match+best_len-1) != scan_end ||
-                       *(bestcmp_t *)(match) != scan_start)) {
-                if ((cur_match = prev[cur_match & wmask]) > limit && --chain_length != 0) {
-                    continue;
-                }
-                cont = 0;
+#ifdef UNALIGNED_OK
+        if (best_len < sizeof(uint32_t)) {
+            for (;;) {
+                match = window + cur_match;
+                if (UNLIKELY(*(uint16_t *)(match+offset) == (uint16_t)scan_end &&
+                             *(uint16_t *)(match) == (uint16_t)scan_start))
+                    break;
+                GOTO_NEXT_CHAIN;
             }
-            break;
-        } while (1);
-
-        if (!cont)
-            break;
-
-#if MIN_MATCH >= 2 && defined(UNALIGNED_OK)
-        len = COMPARE256(scan+2, match+2) + 2;
+        }
```

Comment:
This } could go in front of the else if below, and the same with the next instance below where } can go in front of else, that way we can avoid the infamous "unbalanced else", and improving readability a tiny bit.

Reply:
Fixed.